arch/arm/core/aarch64/switch.S

```diff
@@ -48,6 +48,28 @@ SECTION_FUNC(TEXT, z_arm64_context_switch)
 	mov	x1, sp
 	str	x1, [x2]
 
+#ifdef ___thread_t_tls_OFFSET
+        /* Set thread local storage pointer */
+#if ___thread_t_tls_offset > 255
+	ldr x2, =___thread_t_tls_OFFSET
+	adds x2, x2, x0
+	ldr x2, [x2]
+#else
+	ldr x2, [x0, #___thread_t_tls_OFFSET]
+#endif
+
+	/* __tls needs to point two pointers before storage to
+	 * match ARM ABI for how TLS vars are relocated
+	 */
+#if __SIZE_WIDTH__ == 32
+#define TCB_SIZE 8
+#else
+#define TCB_SIZE 16
+#endif
+	subs x2, x2, #TCB_SIZE
+	msr tpidr_el0, x2
+#endif
+
```

Comment:
Can you use the same spacing and comment style of the rest of the file?

Reply:
Yes, of course, I've pushed an update with the formatting adjusted to match as best I could tell.